homeassistant/helpers/update_coordinator.py

```diff
@@ -141,6 +142,14 @@ def async_update_listeners(self) -> None:
         for update_callback, _ in list(self._listeners.values()):
             update_callback()
 
+    async def async_shutdown(self) -> None:
+        """Cancel any scheduled call, and ignore new runs."""
+        self._shutdown_requested = True
+        if self._unsub_refresh:
+            self._unsub_refresh()
+            self._unsub_refresh = None
```

Comment:
Please break out duplicate code into `_async_unsub_refresh(self) -> None`

This is in `_unschedule_refresh` and `_unsub_refresh`

Reply:
Found a few more instances, so created a preliminary PR: #91573